nautobot/core/models/generics.py

```diff
@@ -25,6 +31,38 @@ class Meta:
         abstract = True
 
 
+class _NautobotTaggableManager(_TaggableManager):
+    def set(self, *tags, through_defaults=None, **kwargs):
+        """
+        Patch model.tags.set() to be backwards-compatible with django-taggit 1.x and forward-compatible with later.
+
+        Both of these approaches are supported:
+
+        - tags.set("tag 1", "tag 2")  # django-taggit 1.x
+        - tags.set(["tag 1", "tag 2"])  # django-taggit 2.x and later
+        """
+        if len(tags) == 1 and not isinstance(tags[0], (self.through.tag_model(), str)):
+            # taggit 2.x+ style, i.e. `set([tag, tag, tag])`
+            tags = tags[0]
+        else:
+            # taggit 1.x style, i.e. `set(tag, tag, tag)`
+            # Note: logger.warning() only supports a `stacklevel` parameter in Python 3.8 and later
+            if sys.version_info >= (3, 8):
+                logger.warning(
+                    "Deprecated `tags.set(%s)` was called, please change to `tags.set(%s)` instead",
+                    ", ".join([repr(tag) for tag in tags]),
+                    list(tags),
+                    stacklevel=2,
+                )
+            else:  # Python 3.7
+                logger.warning(
+                    "Deprecated `tags.set(%s)` was called, please change to `tags.set(%s)` instead",
+                    ", ".join([repr(tag) for tag in tags]),
+                    list(tags),
+                )
```

Comment:
This reads a little funky to me, so I think using variables helps here:

```suggestion
            tags_unpacked = ", ".join([repr(tag) for tag in tags])
            tags_list = list(tags)
            message = "Deprecated `tags.set(%s)` was called, please change to `tags.set(%s)` instead"
            if sys.version_info >= (3, 8):
                logger.warning(message, tags_unpacked, tags_list, stacklevel=2)
            else:  # Python 3.7
                logger.warning(message, tags_unpacked, tags_list)
```